File: cpp/design-patterns/bridge/remote/remote.cpp
class RemoteControl {
protected:
    Device* device;
public:
    RemoteControl(Device* dev) : device(dev) {}
    void togglePower() {
        if (device->isEnabled()) device->disable();
        else device->enable();
    }
    void volumeUp() { device->setVolume(device->getVolume() + 10); }
};

File: go/design-patterns/bridge/remote/remote.go
type RemoteControl struct {
    device Device
}
func (r *RemoteControl) TogglePower() {
    if r.device.IsEnabled() {
        r.device.Disable()
    } else {
        r.device.Enable()
    }
}
type AdvancedRemote struct {
    RemoteControl
}
func (r *AdvancedRemote) Mute() {
    r.device.SetVolume(0)
}